### PR TITLE
[New Rules] GDB Secret Dumping

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
This PR proposes the addition of two new GDB secret dumping rules:
1. Linux init (PID 1) Secret Dump via GDB
2. Linux Secret Dumping via GDB - BBR

## Linux init (PID 1) Secret Dump via GDB
This rule monitors for the potential memory dump of the init process (PID 1) through gdb. Attackers may leverage memory dumping techniques to attempt secret extraction from privileged processes. Tools that display this behavior include "truffleproc" and "bash-memory-dump". This behavior should not happen by default, and should be investigated thoroughly.

### Detection
**0 FPs in telemetry, also 0 FPs in my own stacks. Many FPs of security testing.**
```
process where host.os.type == "linux" and event.action == "exec" and event.type == "start" and 
process.name == "gdb" and process.args == "--pid" and process.args == "1"
```
image

## Linux Secret Dumping via GDB
This rule monitors for potential memory dumping through gdb. Attackers may leverage memory dumping techniques to attempt secret extraction from privileged processes. Tools that display this behavior include "truffleproc" and "bash-memory-dump". This behavior should not happen by default, and should be investigated thoroughly.

### Detection
**0 FPs in telemetry, also 0 FPs in my own stacks. Many FPs of security testing.**
```
process where host.os.type == "linux" and event.action == "exec" and event.type == "start" and 
process.name == "gdb" and process.args == "--pid" and process.args == "1"
```
image

### CODE REVIEW

File: rules/linux/credential_access_gdb_init_memory_dump.toml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+]
+risk_score = 47
+rule_id = "d4ff2f53-c802-4d2e-9fb9-9ecc08356c3f"
+severity = "medium"
+tags = ["Domain: Endpoint", "OS: Linux", "Use Case: Threat Detection", "Tactic: Credential Access", "Data Source: Elastic Defend"]
+timestamp_override = "event.ingested"
+type = "eql"
+query = '''
+process where host.os.type == "linux" and event.action == "exec" and event.type == "start" and 
+process.name == "gdb" and process.args == "--pid" and process.args == "1"
```
Comment (w0rk3r) [inline rules/linux/credential_access_gdb_init_memory_dump.toml:33]: Does `-p` work here too?

File: rules_building_block/credential_access_gdb_memory_dump.toml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+]
+risk_score = 21
+rule_id = "66c058f3-99f4-4d18-952b-43348f2577a0"
+severity = "low"
+tags = ["Domain: Endpoint", "OS: Linux", "Use Case: Threat Detection", "Tactic: Credential Access", "Data Source: Elastic Defend", "Rule Type: BBR"]
+timestamp_override = "event.ingested"
+type = "eql"
+query = '''
+process where host.os.type == "linux" and event.action == "exec" and event.type == "start" and 
+process.name == "gdb" and process.args == "--pid"
```
Comment (w0rk3r) [inline rules_building_block/credential_access_gdb_memory_dump.toml:35]: This one will duplicate the other rule alerts, do you think we should add an exclusion?

File: rules/linux/credential_access_gdb_init_memory_dump.toml (hunk 1 above)
Comment (Aegrah) [inline rules/linux/credential_access_gdb_init_memory_dump.toml:33]:
```suggestion
process.name == "gdb" and process.args in ("--pid", "-p") and process.args == "1"
```

File: rules_building_block/credential_access_gdb_memory_dump.toml (hunk 2 above)
Comment (Aegrah) [inline rules_building_block/credential_access_gdb_memory_dump.toml:35]:
```suggestion
process.name == "gdb" and process.args in ("--pid", "-p") and not process.args == "1"
```

File: rules/linux/credential_access_gdb_init_memory_dump.toml (hunk 1 above)
Comment (Aegrah) [reply rules/linux/credential_access_gdb_init_memory_dump.toml:33]: You sir, are correct

File: rules_building_block/credential_access_gdb_memory_dump.toml (hunk 2 above)
Comment (Aegrah) [reply rules_building_block/credential_access_gdb_memory_dump.toml]: Done

File: rules_building_block/credential_access_gdb_memory_dump.toml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+]
+risk_score = 21
+rule_id = "66c058f3-99f4-4d18-952b-43348f2577a0"
+severity = "low"
+tags = ["Domain: Endpoint", "OS: Linux", "Use Case: Threat Detection", "Tactic: Credential Access", "Data Source: Elastic Defend", "Rule Type: BBR"]
+timestamp_override = "event.ingested"
+type = "eql"
+query = '''
+process where host.os.type == "linux" and event.action == "exec" and event.type == "start" and 
+process.name == "gdb" and process.args in ("--pid", "-p") and not process.args == "1"
```
Comment (Aegrah) [inline rules_building_block/credential_access_gdb_memory_dump.toml:35]:
```suggestion
process.name == "gdb" and process.args in ("--pid", "-p") and process.args != "1"
```

**Review by w0rk3r — APPROVED** [review]

**Review by Samirbous — APPROVED** [review]

**Pull request merged**